### PR TITLE
Feat/change mute command args

### PR DESCRIPTION
**Pull request by watermelon1024**
# Description

Optimize input arguments to better suit user-friendly usage.

Fixes # (issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by a3510377 — APPROVED** [review]

File: lipoic/cogs/admin/mute.py [hunk 1]
```diff
@@ -1,50 +1,144 @@
...
+        super().__init__(bot)
+        self.tz = datetime.timezone(datetime.timedelta(hours=+8))
+
     @commands.has_permissions(moderate_members=True)
     @discord.slash_command(description="Mute Member", guild_only=True)
     async def mute(
         self,
         ctx: ApplicationContext,
-        member: Option(discord.Member, "輸入要禁言的成員(預設時間1分鐘)"),
+        member: Option(discord.Member, "輸入要禁言的成員(預設時間5分鐘)"),
```
Comment (EvanHsieh0415) [inline lipoic/cogs/admin/mute.py:26]: `(預設時間5分鐘)`建議搬到command description裡

File: lipoic/cogs/admin/mute.py [hunk 2]
```diff
@@ -1,50 +1,144 @@
...
+        duration: Option(
+            str, "持續時間(和until衝突)，格式: 天d時h分m秒s，例子: 1d5h10m, 30m", required=False
+        ),
+        until: Option(
+            str,
+            "直到某個時間點(和duration衝突，以台灣時區為主)，格式: 年-月-日 時:分:秒(24小時制)，例子: 2022-04-01 15:30:20, 03-04 20:00",  # noqa
```
Comment (EvanHsieh0415) [inline lipoic/cogs/admin/mute.py:32]: `以台灣時區為主` -> `UTF+8為主`

File: lipoic/cogs/admin/mute.py [hunk 3]
```diff
@@ -1,50 +1,144 @@
...
+        if duration and until:
+            return await ctx.respond(
+                embed=discord.Embed(
+                    title="禁言失敗!", description="duration和until參數不可同時出現!", color=0xE74C3C
+                ),
+                ephemeral=True,
+            )
+
+        if duration:
+            match = re.findall(r"(\d+)([a-z]*)", duration)
```
Comment (EvanHsieh0415) [inline lipoic/cogs/admin/mute.py:46]: regex compile可以預先定義 這樣會加快速度
舉例來說：
```py
compile = re.compile(r"(\d+)([a-z]*)")
re.findall(compile, ...)
```

File: lipoic/cogs/admin/mute.py [hunk 4]
```diff
@@ -1,50 +1,144 @@
...
+                return await ctx.respond(
+                    embed=discord.Embed(
+                        title="禁言失敗!", description="未知的時間格式!", color=0xE74C3C
+                    ),
+                    ephemeral=True,
+                )
+
+            hour = 0
+            minute = 0
+            second = 0
```
Comment (EvanHsieh0415) [inline lipoic/cogs/admin/mute.py:97]:
```py
hour = minute = second = 0
```

File: lipoic/cogs/admin/mute.py [hunk 5]
```diff
@@ -1,50 +1,144 @@
...
+            for num, unit in match:
+                num = int(num)
+                if unit == "d":
+                    delta += datetime.timedelta(days=num)
+                elif unit == "h":
+                    delta += datetime.timedelta(hours=num)
+                elif unit == "m":
+                    delta += datetime.timedelta(minutes=num)
+                elif unit == "s":
+                    delta += datetime.timedelta(seconds=num)
```
Comment (EvanHsieh0415) [inline lipoic/cogs/admin/mute.py:57]: 可以嘗試改成
```py
maps = {'d': 'days', ...}
delta += datetime.timedelta(**{maps[unit]: num})
```

**Review by EvanHsieh0415 — COMMENTED** [review]
LGTM

**Review by HansHans135 — APPROVED** [review]

**Review by EvanHsieh0415 — APPROVED** [review]
LGTM

**Pull request merged**